cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java

```diff
@@ -72,13 +74,16 @@ public DynamicMapReduceContext(Program program,
                                  RunId runId, String taskId,
                                  Arguments runtimeArguments,
                                  MapReduceSpecification spec,
-                                 long logicalStartTime, String workflowBatch, String adapterName,
+                                 long logicalStartTime, String workflowBatch,
                                  DiscoveryServiceClient discoveryServiceClient,
                                  MetricsCollectionService metricsCollectionService,
                                  TransactionSystemClient txClient,
-                                 DatasetFramework dsFramework) {
+                                 DatasetFramework dsFramework,
+                                 @Nullable AdapterSpecification adapterSpec,
```

Comment:
nit: Looks like ``type`` and ``taskId`` are also ``@Nullable``. Unrelated, but if it doesn't increase the size of this PR, maybe we can mark them so as well?